src/views/Token/index.tsx

```diff
@@ -65,45 +67,44 @@ query getTokenClaims {
 
 `
 
-// { data: buyer { saleId:.., tokens}}
-
 export function TokenView() {
   const dispatch = useDispatch()
+  const { account, library, chainId } = useWeb3React()
   const [t] = useTranslation()
   const [loading, setLoading] = useState<boolean>(true)
   const [error, setError] = useState<Error>()
   const mesa = useMesa()
-  const [filteredData, setFilteredData] = useState<FixedPriceSalePurchase[] | undefined>()
-  const account = (window as any).ethereum.selectedAddress
-
-  //  const fetchdata = async () => {
-  //    try {
-  //      const data = await mesa.subgraph.query(userSalesQuery(account))
-  //      const unixDateNow = dayjs(Date.now()).unix()
-  //      const filteredData = data.fixedPriceSalePurchases.filter(
-  //        (element: FixedPriceSalePurchase) =>
-  //          BigNumber.from(element.sale?.soldAmount) >= BigNumber.from(element.sale?.minimumRaise) &&
-  //          unixDateNow > element.sale!.endDate
-  //      )
-  //      return filteredData
-  //    } catch (error) {
-  //      setError(error)
-  //      setLoading(false)
-  //    }
-  //  }
-  //
+  const [filteredData, setFilteredData] = useState<any[] | undefined>()
+
   useEffect(() => {
+    if (!account || !library || !chainId) {
+      return
+    }
+
     mesa.subgraph
       .query(userSalesQuery(account))
       .then(({ data }) => {
         const { fixedPriceSalePurchases } = data
         const unixDateNow = dayjs(Date.now()).unix()
+        const dataArray = fixedPriceSalePurchases.filter(
+          (element: FixedPriceSalePurchase) =>
+            BigNumber.from(element.sale?.soldAmount) >= BigNumber.from(element.sale?.minimumRaise) &&
+            unixDateNow > element.sale!.endDate
+        )
         setFilteredData(
-          fixedPriceSalePurchases.filter(
-            (element: FixedPriceSalePurchase) =>
-              BigNumber.from(element.sale?.soldAmount) >= BigNumber.from(element.sale?.minimumRaise) &&
-              unixDateNow > element.sale!.endDate
-          )
+          _(dataArray)
+            .groupBy(purchase => purchase.sale?.id)
+            .map((purchase: any) => ({
+              sale: purchase[0].sale,
+              amount: _.reduce(
+                purchase,
+                (total, n) => {
+                  return BigNumber.from(total).add(BigNumber.from(n.amount))
+                },
+                BigNumber.from(0)
+              ),
+            }))
+            .value()
```

Comment:
The `FixedPriceSaleUser` can have totalPurchaseAmount` to aggregate this. `groupBy` is also easy to implement with reduce and `Record` type. Remove `lodash`